### PR TITLE
Added copy constructor for various built-in types.

### LINKED ISSUE #24001: Vector2(position) constructor causes an error
Godot 3.1 alpha2

I get an error when I try to construct a `Vector2` from another one:
```gdscript
extends Node2D


func _draw():
	var pos = self.position
	var angle = PI
	draw_set_transform(Vector2(pos), angle, Vector2(1,1))
```

Causes an error at the last line:
```
SCRIPT ERROR: GDScript::reload: Parse Error: No constructor of 'Vector2' matches the signature 'Vector2(var)'.
          At: built-in:7
```

This code worked in Godot 3.0.6, although I have to admit it can be fairly simplified :p
The same happens with Vector3.

### PR DESCRIPTION
**Pull request by vinterskog**
_bugsquad edit_: Fixes #24001

### CODE REVIEW

**Comment by akien-mga** [comment]
There are build and style issues: https://travis-ci.org/godotengine/godot/builds/461693498

**Comment by akien-mga** [comment]
See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) for instructions on how to rebase your branch properly to remove the unrelated commits (and eventually squash the fixups).

**Comment by vinterskog** [comment]
Sorry, having a really hard time with GIT here. I was following the workflow and have absolutely no idea whatsoever what happened. Never used GIT before.

File: core/math/vector3.h [hunk 1]
```diff
@@ -143,6 +143,11 @@ struct Vector3 {
 	operator String() const;
 
 	_FORCE_INLINE_ Vector3() { x = y = z = 0; }
+	_FORCE_INLINE_ Vector3(const Vector3 &v2) {
```
Comment (aaronfranke) [inline core/math/vector3.h]: This definitely shouldn't be called `v2`, since it's a Vector3

File: core/color.h [hunk 2]
```diff
@@ -210,6 +210,16 @@ struct Color {
 		a = 1.0;
 	}
 
+	/**
+	 * Copy constructor
+	 */
+	_FORCE_INLINE_ Color(const Color &c2) {
```
Comment (aaronfranke) [inline core/color.h]: Why is everything `c2` etc? It'd be better to call it `p_c` etc.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: core/color.h (hunk 2 above)
Comment (vinterskog) [reply core/color.h]: Sure thing, I'll fix that.

**Comment by akien-mga** [comment]
Could you squash commits together? See [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

File: core/variant_call.cpp [hunk 3]
```diff
@@ -1854,34 +1894,44 @@ void register_variant_methods() {
...
+	_VariantCall::add_constructor(_VariantCall::Rect2_init1, Variant::RECT2, "rect2", Variant::RECT2);
+	_VariantCall::add_constructor(_VariantCall::Rect2_init2, Variant::RECT2, "position", Variant::VECTOR2, "size", Variant::VECTOR2);
+	_VariantCall::add_constructor(_VariantCall::Rect2_init3, Variant::RECT2, "x", Variant::REAL, "y", Variant::REAL, "width", Variant::REAL, "height", Variant::REAL);
 
+	_VariantCall::add_constructor(_VariantCall::Transform2D_init1, Variant::TRANSFORM2D, "t2", Variant::TRANSFORM2D);
 	_VariantCall::add_constructor(_VariantCall::Transform2D_init2, Variant::TRANSFORM2D, "rotation", Variant::REAL, "position", Variant::VECTOR2);
 	_VariantCall::add_constructor(_VariantCall::Transform2D_init3, Variant::TRANSFORM2D, "x_axis", Variant::VECTOR2, "y_axis", Variant::VECTOR2, "origin", Variant::VECTOR2);
 
-	_VariantCall::add_constructor(_VariantCall::Vector3_init1, Variant::VECTOR3, "x", Variant::REAL, "y", Variant::REAL, "z", Variant::REAL);
+	_VariantCall::add_constructor(_VariantCall::Vector3_init1, Variant::VECTOR3, "v2", Variant::VECTOR3);
```
Comment (aaronfranke) [inline core/variant_call.cpp]: These `_VariantCall` items still contain the names `v2` etc.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

**Comment by akien-mga** [comment]
Moving to 3.2 milestone as we're about to enter the beta stage and *release freeze* for Godot 3.1. Only bug fixes against issues of the 3.1 milestone (or enhancements to core features of the 3.1 roadmap) will be considered until 3.1-stable is released.

**Comment by aaronfranke** [comment]
This is most likely entirely superseded by #34948, so this should probably be closed. @vinterskog If there is anything else in this PR that is still necessary, let us know.

**Comment by aaronfranke** [comment]
Closing as superseded by #34948.

**Pull request closed without merging**